### PR TITLE
[WIP] Fix permission screen overflow

### LINKED ISSUE #13506: [Bug]: Permissions request: account selection screen has odd overlaps
### Describe the bug

The permissions request UI is buggy when a user has many accounts.

### Steps to reproduce

1. Have a long list of accounts in MetaMask.  I recommend at least a dozen; demo observed with ~40. 
2. Visit https://metamask.github.io/test-dapp/
3. In the "Permissions Actions" box, click "Request Permissions." 
4. MetaMask opens with a screen labeled in the upper right "1 of 2" titled "Connect With MetaMask: Select the account(s) to use on this site."  There should be one scrollbar on the left. 
5. Select one account near the top of the list to connect with, and scroll down. 

Actual behavior: 
The text "Only connect with sites you trust. Learn more" overlaps the bottom of the box for the 8th account; the "Cancel" and "Next" buttons overlap the 9th and 10th accounts (making the 9th non-selectable; you can't connect with that one).  There are no buttons/text at the bottom of the list; you just have to know to look around the 9th/10th account for buttons to click to proceed.

Expected behavior: 
The text and buttons come after the last account in the list.

### Error messages or log output

_No response_

### Version

10.9.0

### Build type

_No response_

### Browser

Chrome

### Operating system

Windows

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GuillaumeRx**
Fixes: #13506

Explanation:  Fixes permission screen issues by re-making the account list scrollable

Manual testing steps:  
  - Have a long list of accounts (~15)
  - Try to connect an account
  - Look at the list

### CODE REVIEW

File: ui/components/ui/account-list/index.scss [hunk 1]
```diff
@@ -32,15 +32,16 @@
   &__header-check-box {
     margin-right: 16px;
   }
-
+
```
Comment (rekmarks) [inline ui/components/ui/account-list/index.scss:35]:
```suggestion

```

**Review by rekmarks — APPROVED** [review]
LGTM!

**Comment by rekmarks** [comment]
(The noisy commit activity is due to me trying and failing to do f12e85a using the suggestion editor in the files view of this PR and then having to fix it locally.)

**Comment by darkwing** [comment]
@georgewrmarshall Were you able to make progress here?

**Comment by georgewrmarshall** [comment]
@darkwing sorry I did not! Looking into it now

**Comment by georgewrmarshall** [comment]
@darkwing looked into it and we might be able to fix using a combination of diplay:grid for the whole page and overflow auto. This will take a bit of time and testing. So my suggestion would be to merge this as a hot fix and create a new issue to implement and improved fix.

**Review by georgewrmarshall — APPROVED** [review]
lgtm!

**Pull request merged**